presto-docs/src/main/sphinx/security/hive.rst

```diff
@@ -0,0 +1,237 @@
+list of properties that are involved in HDFS authentication is given below;
+a complete discussion of the modes follows.
+
+================================================== ============================================================ ==========
+Property Name                                      Description                                                  Default
+================================================== ============================================================ ==========
+``hive.hdfs.authentication.type``                  One of ``SIMPLE`` or ``KERBEROS``. When set to ``KERBEROS``, ``SIMPLE``
+                                                   the Hive connector authenticates to HDFS using Kerberos.
+                                                   When using the default value of ``SIMPLE``, Kerberos
+                                                   authentication is disabled and no other properties need to
+                                                   be configured.
+
+``hive.hdfs.presto.principal``                     When ``hive.hdfs.authentication.type`` is set to
+                                                   ``KERBEROS``, the Kerberos principal that Presto will use
+                                                   when connecting to HDFS.
+
+``hive.hdfs.presto.keytab``                        The path to the keytab file that contains a key for the
+                                                   principal specified by hive.hdfs.presto.principal.
+
+================================================== ============================================================ ==========
+
+.. _hive-security-simple:
+
+``SIMPLE``
+^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+
+When using ``SIMPLE`` authentication, Presto accesses HDFS as the user the
+Presto process is running as. This is the default authentication type, and no
+additional properties need to be set. Kerberos is not used.
+
+.. _hive-security-kerberos:
+
+``KERBEROS``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=KERBEROS
+    hive.hdfs.presto.principal=hdfs@EXAMPLE.COM
+    hive.hdfs.presto.keytab=/etc/presto/hdfs.keytab
+
+When using ``KERBEROS`` authentication, Presto accesses HDFS as the principal
+specified by the ``hive.hdfs.presto.principal`` property. Presto authenticates
+this principal using the keytab specified by the ``hive.hdfs.presto.keytab``
+keytab.
+
+Keytab files must be distributed to every node in the cluster.
+
+:ref:`Additional information on keytab files.<hive-security-additional-keytab>`
+
+End user impersonation
+======================
+
+Impersonation Accessing HDFS
+----------------------------
+
+Presto can impersonate the end user who is running a query. In the case of a
+user running a query from the command line interface, this is the value of the
+``--user`` option. Impersonating the end user can provide additional security
```

Comment:
This is actually the value of System.getProperty("user.name") unless --user is used to override that. 

See:
https://github.com/facebook/presto/blob/master/presto-cli/src/main/java/com/facebook/presto/cli/ClientOptions.java#L71